From 249a6b8d19d738f067f6e424d706919f7c0bdba6 Mon Sep 17 00:00:00 2001
From: Algea Cao <algea.cao@rock-chips.com>
Date: Wed, 26 Dec 2018 10:02:20 +0800
Subject: [PATCH] drm/bridge: synopsys: dw-hdmi: Remove hpd_state judgment of
 atomic_begin and atomic_flush

There is no need to judge hpd status in atomic_begin and atomic_flush.
And this judgment may cause display error if TV make hpd status change
frequently.

Change-Id: I2ed87ef42b78a8faadc4bcc5b6b16d9390644903
Signed-off-by: Algea Cao <algea.cao@rock-chips.com>
---
 drivers/gpu/drm/bridge/synopsys/dw-hdmi.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/bridge/synopsys/dw-hdmi.c b/drivers/gpu/drm/bridge/synopsys/dw-hdmi.c
index f1222cfa6bd6..0cdda3112084 100644
--- a/drivers/gpu/drm/bridge/synopsys/dw-hdmi.c
+++ b/drivers/gpu/drm/bridge/synopsys/dw-hdmi.c
@@ -2585,7 +2585,7 @@ dw_hdmi_connector_atomic_begin(struct drm_connector *connector,
 	unsigned int enc_in_encoding;
 	unsigned int enc_out_encoding;
 
-	if (!hdmi->hpd_state || !conn_state->crtc)
+	if (!conn_state->crtc)
 		return;
 
 	if (!hdmi->hdmi_data.video_mode.mpixelclock)
@@ -2636,7 +2636,7 @@ dw_hdmi_connector_atomic_flush(struct drm_connector *connector,
 	unsigned int out_bus_format = hdmi->hdmi_data.enc_out_bus_format;
 
 
-	if (!hdmi->hpd_state || !conn_state->crtc)
+	if (!conn_state->crtc)
 		return;
 
 	DRM_DEBUG("%s\n", __func__);
-- 
2.35.3

